src/modules/transaction/components/TxSignatureCollector/TxSignatureCollector.js

```diff
@@ -108,12 +108,14 @@ const TxSignatureCollector = ({
   };
 
   const onEnterPasswordSuccess = ({ privateKey }) => {
-    const paramsSchema = moduleCommandSchemas[moduleCommand];
-    const transaction = fromTransactionJSON(transactionJSON, paramsSchema);
-    const buffer = encodeTransaction(transaction, paramsSchema);
-    const transactionHex = cryptography.utils.hash(buffer).toString('hex');
-    if (isTransactionAuthor) {
-      incrementNonce(transactionHex);
+    if (transactionJSON.nonce) {
```

Comment:
I think this was not the bug? we need to find the root cause?

it could be the `isTransactionAuthor` which is declared in global scope? please check the root of the problem and fix

Reply:
Is this one related to signing a message? I think it is a good idea to move that logic inside the if statement because it is not used outside of the if. However, should we not still use `if(isTransactionAuthor)`